server/src/main/java/org/opensearch/index/query/BoolQueryBuilder.java

```
@@ -152,6 +152,19 @@ public List<QueryBuilder> filter() {
         return this.filterClauses;
     }
 
+    /**
+     * Add query filter to filterClauses.
+     * @param filter the query filter
+     * @param filterCombinationMode the filter combination mode
+     * @return BoolQueryBuilder
+     */
+    public BoolQueryBuilder filter(QueryBuilder filter, FilterCombinationMode filterCombinationMode) {
+        if(FilterCombinationMode.AND.equals(filterCombinationMode) && filter != null) {
```

I think by default if the combination mode is null we will use AND logic to combine them. I think it's better to add a check for unknown combination mode and throw unsupported operation exception in case later people add a new mode but forget to implement it for each query builder. With that exception we can catch the issue rather than silently do nothing.

---

Sounds good!